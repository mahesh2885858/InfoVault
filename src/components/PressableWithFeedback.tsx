import React from 'react';
import {
  ColorValue,
  Pressable,
  PressableProps,
  StyleProp,
  ViewStyle,
} from 'react-native';

const PressableWithFeedback = (
  props: PressableProps & {feedbackColor?: ColorValue; hidden?: boolean},
) => {
  const {children, feedbackColor} = props;
  if (props.hidden) return null;
  return (
    <Pressable
      {...props}
      style={({pressed}) => [
        {
          backgroundColor: pressed ? feedbackColor : 'transparent',

          opacity: pressed ? 0.5 : 1,
        },
        props.style as StyleProp<ViewStyle>,
      ]}>
      {children}
    </Pressable>
  );
};

export default PressableWithFeedback;
