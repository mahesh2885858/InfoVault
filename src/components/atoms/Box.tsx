import React from 'react';
import {View, ViewProps} from 'react-native';
import {myTheme} from '../../../theme';

const Box = (props: ViewProps) => {
  return (
    <View
      {...props}
      style={[{backgroundColor: myTheme.secondary}, props.style]}>
      {props.children}
    </View>
  );
};

export default Box;
