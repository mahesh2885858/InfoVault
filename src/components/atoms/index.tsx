export { default as Container } from './Container';
export { default as Typography } from './Typography';
export { default as Button } from './Button';
export { default as Box } from './Box';
