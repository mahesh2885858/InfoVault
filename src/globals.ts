export const colors = {
  background: '#0B141A',
  backgroundLight: '#242C31',
  primaryGreen: '#21C063',
  secondaryText: '#8696A0',
  primaryText: '#ffffff',
  messageBgOther: '#202C33',
  messageLink: '#53BDEB',
  messageBgSelf: '#005C4B',
  messageRead: '#50B9E5',
};

export const sizes = {
  xxs: 1,
  xs: 2,
  s: 4,
  sm: 6,
  m: 8,
  ml: 12,
  l: 16,
  lxl: 20,
  xl: 24,
  xxl: 30,
  xxxl: 36,
};
