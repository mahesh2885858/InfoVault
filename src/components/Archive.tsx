import React from 'react';
import {Pressable, StyleSheet, Text, View} from 'react-native';
import MaterialIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import {colors} from '../globals.ts';

const Archive = (props: {
  setArchiveMode: React.Dispatch<React.SetStateAction<boolean>>;
}) => {
  return (
    <>
      <Pressable
        style={styles.container}
        onPress={() => {
          props.setArchiveMode(true);
        }}>
        <View style={styles.icon}>
          <MaterialIcon
            name="archive-arrow-down-outline"
            color={colors.primaryText}
            size={24}
          />
        </View>
        <Text style={styles.text}>Archived</Text>
      </Pressable>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
    // paddingLeft: 25,
    display: 'flex',
    flexDirection: 'row',
    gap: 15,
    alignItems: 'center',
  },
  icon: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 50,
    width: 50,
  },
  text: {
    color: colors.primaryText,
    fontWeight: '700',
    fontSize: 14,
  },
});

export default Archive;
