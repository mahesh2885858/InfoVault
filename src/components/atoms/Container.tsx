import React from 'react';
import {View, ViewProps} from 'react-native';
import {useTheme} from '@ui-kitten/components';

const Container = (props: ViewProps) => {
  const theme = useTheme();
  return (
    <View {...props} style={[{backgroundColor: theme['bg-main']}, props.style]}>
      {props.children}
    </View>
  );
};

export default Container;
