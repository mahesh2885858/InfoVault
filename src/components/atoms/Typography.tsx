import React from 'react';
import {useTheme} from '@ui-kitten/components';
import {Text, type TextProps} from '@ui-kitten/components';

const Typography = (props: TextProps) => {
  const theme = useTheme();
  return (
    <Text {...props} style={[{color: theme['text-primary']}, props.style]}>
      {props.children}
    </Text>
  );
};

export default Typography;
