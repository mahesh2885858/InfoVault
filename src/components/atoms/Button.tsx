import React from 'react';
import PressableWithFeedback from '../PressableWithFeedback';
import Typography from './Typography';
import {StyleService, useStyleSheet} from '@ui-kitten/components';

type TProps = {
  onButtonPress: () => void;
  label: string;
};

const Button = (props: TProps) => {
  const styles = useStyleSheet(themedStyles);
  return (
    <PressableWithFeedback
      style={styles.button}
      onPress={() => props.onButtonPress()}>
      <Typography style={styles.text}>{props.label}</Typography>
    </PressableWithFeedback>
  );
};

const themedStyles = StyleService.create({
  button: {
    paddingHorizontal: 15,
    paddingVertical: 7,
    borderRadius: 5,
    backgroundColor: 'button-primary-bg',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 16,
    color: 'button-primary-text',
  },
});

export default Button;
