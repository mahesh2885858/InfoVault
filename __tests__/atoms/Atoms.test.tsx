import 'react-native';
import Typography from '../../src/components/atoms/Typography';
import { it } from '@jest/globals';
import React from 'react';
import Box from '../../src/components/atoms/Box';
import Button from '../../src/components/atoms/Button';
import Container from '../../src/components/atoms/Container';
import PressableWithFeedback from '../../src/components/PressableWithFeedback';
import { fireEvent, render, screen } from '@testing-library/react-native';
import * as eva from '@eva-design/eva';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import { ApplicationProvider, IconRegistry } from '@ui-kitten/components';
import { SafeAreaProvider } from 'react-native-safe-area-context';

// Test wrapper component
const TestWrapper = ({ children }: { children: React.ReactNode }) => (
  <SafeAreaProvider>
    <IconRegistry icons={EvaIconsPack} />
    <ApplicationProvider {...eva} theme={eva.light}>
      {children}
    </ApplicationProvider>
  </SafeAreaProvider>
);

it('renders Light Text correctly', () => {
  render(
    <TestWrapper>
      <Typography />
    </TestWrapper>,
  );
});

it('renders Box correctly', () => {
  render(<Box />);
});

it('renders Pressasble correctly', () => {
  const mockProps = {
    feedbackColor: 'red',
    hidden: false,
    style: { padding: 10 },
    onPress: jest.fn(),
  };
  const tree = render(<PressableWithFeedback {...mockProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Hide Pressasble correctly', () => {
  const mockProps = {
    feedbackColor: 'red',
    hidden: true,
    style: { padding: 10 },
    onPress: jest.fn(),
  };
  const tree = render(<PressableWithFeedback {...mockProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Button correctly', () => {
  const mockProps = {
    label: 'test',
    onButtonPress: jest.fn(),
  };
  render(
    <TestWrapper>
      <Button {...mockProps} />
    </TestWrapper>,
  );
  fireEvent(screen.getByText('test'), 'press');
  expect(mockProps.onButtonPress).toHaveBeenCalledTimes(1);
});

it('renders Container correctly', () => {
  const tree = render(
    <TestWrapper>
      <Container />
    </TestWrapper>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
