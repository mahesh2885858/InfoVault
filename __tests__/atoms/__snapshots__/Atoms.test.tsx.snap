// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hide Pressasble correctly 1`] = `null`;

exports[`renders Container correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": undefined,
      },
      undefined,
      {
        "paddingBottom": 0,
        "paddingTop": 0,
      },
    ]
  }
/>
`;

exports[`renders Pressasble correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  feedbackColor="red"
  focusable={true}
  hidden={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "transparent",
        "opacity": 1,
      },
      {
        "padding": 10,
      },
    ]
  }
/>
`;
